sdk-base/src/main/java/com/mapbox/maps/plugin/viewport/transition/ViewportTransition.kt

```diff
@@ -0,0 +1,31 @@
+package com.mapbox.maps.plugin.viewport.transition
+
+import com.mapbox.maps.plugin.animation.Cancelable
+import com.mapbox.maps.plugin.viewport.CompletionListener
+import com.mapbox.maps.plugin.viewport.state.ViewportState
+
+/**
+ * Defines how a [ViewportState] transition to another [ViewportState].
+ */
+interface ViewportTransition {
```

Comment:
could be `fun interface`?

Reply:
🤔 I'm not sure as this interface is not supposed to be used as a listener which SAM would be more useful. It's more used as an interface where user's complex transition logic class conforms, but would like @mapbox/maps-android 's thoughts as well.

Reply:
added it to my todo list between beta and rc. will think more about it.